### DISCUSSION TITLE
How to 'stack' multiple data tables under single New-HTMLContent

### DISCUSSION THREAD

**Discussion by ComputerRecon** [EvotecIT/PSWriteHTML#322] [Q&A] [answered]
I have tried many, many different variations and combinations, but I just can't seem to get two data tables to stack on top of each other when buried in a ForEach statement to create new-HTMLContent.  They always align side by side.

My code is (will try to shorten it, but can't seem to get formatting to work right on here):

```powershell
New-HTMLContent -HeaderText "Citrix Machine Information" -HeaderTextSize "16" -HeaderTextColor "#ffffff" -HeaderBackGroundColor "#383392" -BackgroundColor "#707cf1" -AlignContent center -AlignItems center -CanCollapse {
try
{				
	$CtxMachine01Info = @()
	$CtxMachine02Info = @()
	$CtxMachineInfo = Invoke-Command -Computer $Computer {Get-BrokerMachine | Select * | Sort -Property CatalogName, DNSName} | Select -Property * -ExcludeProperty PSComputerName, RunSpaceID, PSShowComputerName
				
foreach ($Machine in $CtxMachineInfo)
{
	New-HTMLContent -HeaderText $Machine.DNSName -HeaderTextSize "16" -HeaderTextColor "#ffffff" -HeaderBackGroundColor "#383392" -BackgroundColor "#707cf1" -AlignContent center -AlignItems center -CanCollapse -Collapsed {
						
	$CtxMach01Obj = New-Object PSObject
						
	$CtxMach01Obj | Add-Member NoteProperty -Name "Name" -Value $CtxMachineInfo.DNSName
	$CtxMach01Obj | Add-Member NoteProperty -Name "Machine Catalog" -Value CtxMachineInfo.CatalogName
											
	$CtxMachine01Info += $CtxMach01Obj
	$CtxMach02Obj = New-Object PSObject
						
	$CtxMach02Obj | Add-Member NoteProperty -Name "Broker" -Value $CtxMachineInfo.ControllerDNSName
	$CtxMach02Obj | Add-Member NoteProperty -Name "Registration State" -Value $CtxMachineInfo.RegistrationState
	$CtxMachine02Info += $CtxMach02Obj
	
	New-HTMLTable -DataTable $CtxMachine01Info -HideFooter -HideButtons -DisableSearch
	
	New-HTMLTable -DataTable $CtxMachine02Info -HideFooter -HideButtons -DisableSearch
}

Write-Color "n     $Computer ", "Citrix Machine Data Collection ", "Complete.     n" -Color Cyan, White, Green
}
```

This is how it always looks:
![image](https://user-images.githubusercontent.com/71337273/172861246-dc325b43-b6a4-4411-bfa0-d9b6c69ab087.png)

As mentioned, I've tried so many different combinations and trying to wrap the New-HTMLTable entries in varies Panels/Content etc, but just can't seem to get the two data tables to stack on top of each other.

Anyone have any ideas?  Any help is most appreciated.

**Comment by PrzemyslawKlys** [comment]
```powershell
New-HTML {
    New-HTMLSection {
        $Machines = @(
            'Test1', 'Test2'
        )
        foreach ($Machine in $Machines) {
            New-HTMLSection -HeaderText $Machine {
                New-HTMLTable -DataTable $Machine
            }
        }
    } -HeaderText "Citrix Machine Information" -Direction column
} -Online -ShowHTML
```

`New-HTMLSection` aka `New-HTMLContent` has a parameter called `Direction` which can force things your way.

**Comment by ComputerRecon** [comment]
Thank you for the reply.  I tried to use -Direction column but it didn't seem to have any effect.  In your example, you've only got a single HTMLTable, but I've got two data tables that I'm trying to stack on top of each other within the foreach statement.  It still puts both tables side by side instead of on top of each other.

**Reply by PrzemyslawKlys** [reply]
There are 2 tables in my example. Notice there is a loop. You can have as many as you like. It will work.

**Comment by ComputerRecon** [comment]
Okay, I see what you're saying.  But in your example, you have two machines, and each is getting it's own section within the main 'Citrix Machine Info' section.  That part is working as expected in my example.  So if I have 10 machines, each machine is getting it's own section.
BUT, what I can't get to work, and what I don't think is showing in your example, is that for each machine, I have two data tables ($CtxMachine01Info and $CtxMachine02Info).   It's the two MachineInfo tables that keep aligning side by side instead of stacked on top of each other.

**Comment by PrzemyslawKlys** [comment, marked as answer]
```powershell
New-HTML {
    New-HTMLSection {
        $Machines = @(
            'Machine1', 'Machine2', 'Machine3'
        )
        foreach ($Machine in $Machines) {
            New-HTMLSection -HeaderText $Machine {
                $Information = @(
                    'First Information', '2nd Information', '3rd information'
                )
                foreach ($I in $Information) {
                    New-HTMLSection -HeaderText $I {
                        New-HTMLTable -DataTable $Machine
                    }
                }
            } -Direction column
        }
    } -HeaderText "Citrix Machine Information" -Direction column
} -Online -ShowHTML
```

Works for me

![image](https://user-images.githubusercontent.com/15063294/172897359-c82e716a-4784-494c-99f8-1e46ca92b9fd.png)
